Add Fujicoin

fujicoin opened the pull request:
Please add Fujicoin.

martinboehm commented:
Hi, your PR contains only the config file. See for example [this](https://github.com/trezor/blockbook/pull/79) PR as a reference of coin implementation.
Also, if possible, please use ports xx48, the ports xx47 is already used by another coin.

fujicoin commented:
Thank you for your comment.
I added the necessary files and changed the port number to xx48.
By the way, how should I make testTxPacked1?
https://github.com/fujicoin/blockbook/blob/07300c6b2c3eb8ccdfc938e921d3bd260bb538a1/bchain/coins/fujicoin/fujicoinparser_test.go#L84

fujicoin commented:
![screenshot_2019-01-13 trezor fujicoin explorer](https://user-images.githubusercontent.com/7656623/51083779-f075da00-1762-11e9-9ba0-357f0c3d0976.png)

Deployment in manual build on test environment (ubuntu 16.04) is completed. Function and various display contents are normal. The broadcast of the transaction is also normal.

fujicoin commented:
Deployment in production environment is completed.
The URL is as follows:
https://explorer.fujicoin.org/

fujicoin commented:
![screenshot_2019-01-14 trezor wallet](https://user-images.githubusercontent.com/7656623/51091265-c1478300-17cb-11e9-8b23-c03c21760b50.png)

However, using 'https://explorer.fujicoin.org/' will not connect from the TREZOR wallet.
'https: // 192.168.xx.xx: 9148' is used to connect from the TREZOR wallet.
Is something wrong with the setting of apache2?

fujicoin commented:
![screenshot_2019-01-14 trezor wallet 1](https://user-images.githubusercontent.com/7656623/51097074-99c0dc80-1804-11e9-8ccd-968b5e2318d1.png)

The third point is settled.
Added the following to the proxy setting of apache2.
`
ProxyPass /socket.io/ wss://localhost:9148/socket.io/
ProxyPassReverse /socket.io/ wss://localhost:9148/socket.io/
`

fujicoin commented:
point 1:
I added fujicoin-releases.asc.
point 2:
I am not a go programmer, so I do not know how to pack.

fujicoin commented:
![screenshot_2019-01-14 trezor wallet 2](https://user-images.githubusercontent.com/7656623/51098473-e5c44f00-180d-11e9-9fa9-42c5faf62280.png)

I used a TREZOR wallet to test for deposits and withdrawals. The result is good.

martinboehm commented:
Packing: use this string as packedTx and the test will pass:
`0023f3648bc2d3dd5c0100000001ec5666d4e842926c6e872fd7f12a87183b298464716dec1b03fc956777e06a70010000006a473044022062949ffe184af6bdc8d77e07c2caf1bd4e36d977bc27e24ab6d24c9e735d76b802201f42950f6fa005126f1fbd5cb9d6f0838d94adb70a05517b3eebd144a0ec2fd6012102534f95a0a13bbc3e261f966edd057a1dc9819a4e6bb21d81a633d5500d184a7bfeffffff0200e876481700000017a9146b8a69f8b43bc06b886c0668c93197be5152e8598740494d552e0000001976a914df783f980b38d7a64cee23202af95609b547fde888ac62f32300`

wakiyamap commented on tests/sync/testdata/fujicoin.json:
```diff
@@ -0,0 +1,385 @@
+                            },
+                            {
+                                "value": 385.54800875,
+                                "n": 6
```
```suggestion
                                "n": 6,
```

fujicoin commented:
@martinboehm @wakiyamap tnx, I fixed it.

martinboehm commented:
I am able to run Fujicoin Blockbook, however I cannot merge your PR as the integration tests are failing. What problems I see (there may be more):
1) In the `tests/rpc/testdata/fujicoin.json` are wrong txids in the `blockTxs`
2) In the `tests/sync/testdata/fujicoin.json` are wrong height numbers in the `fakeBlocks` and `realBlocks `

Please make the integration tests working (see the [test documentation](https://github.com/trezor/blockbook/blob/master/docs/testing.md) for more info).

fujicoin commented:
I fixed the pointed out part. I tried testing myself to find out more bugs, but it does not work. I'm sorry but please run the test again.

fujicoin commented:
Finally I can run the test. I corrected the test files and passed all the tests.

fujicoin commented:
@martinboehm Thank you for your kind support. I think that this repository is wonderful. I pray for the development of TREZOR.
